theano/tensor/elemwise.py

```diff
@@ -870,6 +870,13 @@ def perform(self, node, inputs, output_storage):
             elif (not isinstance(variable, numpy.ndarray) or
                   variable.dtype != nout.dtype):
                 variable = numpy.asarray(variable, nout.dtype)
+                # The next line is needed for numpy 1.9. Otherwise
+                # there are tests that fail in DebugMode.
+                # Normally we would call theano.misc._asarray, but it
+                # is faster to inline the code. We know that the dtype
+                # are the same string, just different typenum.
+                if numpy.dtype(nout.dtype).num != variable.dtype.num:
+                    variable = variable.view(dtype=nout.dtype)
```

Comment:
I prefer not to add another check for the dtype. We already passed on line 872 the dtype we wanted, so it must be good. The extra check would slow down that computation and I already optimized it in the past as it was slow on some tests.

If you insist, I can add it, but I would prefer not to do it.